### ISSUE TITLE
dom0 and OpenBSD HVM cursor mismatch

### ISSUE THREAD

**Issue by DemiMarie** [QubesOS/qubes-issues#5387]
### Qubes OS version:

`Qubes release 4.0 (R4.0)`

### Affected component(s) or functionality:

HVM emulated graphics

---

### Steps to reproduce the behavior:

Install OpenBSD in an HVM and enable `xenodm` with `rcctl`.

### Expected or desired behavior:

In-VM cursor and dom0 cursor stay at the same place

### Actual behavior:

They don’t.

### General notes:

Not sure if this is an OpenBSD bug or a QubesOS bug.

---

### I am aware of the following related, [non-duplicate](https://www.qubes-os.org/doc/reporting-bugs/#new-issues-should-not-be-duplicates-of-existing-issues) issues:

#4551

**Comment by marmarek** [comment]
Is it just delay? Constant offset? Or something else?

**Comment by DemiMarie** [comment]
If I vertically move the cursor off of the VM, the offset is zero at the border through which the cursor exited the VM.  It increases linearly as the cursor moves away from that border.

**Comment by DemiMarie** [comment]
I forgot to mention it earlier, but the VM seems to think the screen is larger than it is.

**Comment by DemiMarie** [comment]
Workaround: `xrandr --output default --mode 1280x720` in the AppVM.

**Comment by irelativism** [comment]
o/ qubes, you should consider the HyperbolaBSD project once its finished, a openbsd kernel and userspace hard fork, with the focus in modularity and freedom, so by consequence security and privacy, also all mergerd works will be licensed under GPLv3. other efforts will be included like features from drangonfly, Gentoohardenedbsd debiankfreebsd, pacmanbsd and libertybsd scripts. hyperbola.info for more info. Cheers.

**Comment by rapenne-s** [comment]
> Workaround: `xrandr --output default --mode 1280x720` in the AppVM.

The fix is the right one.

Also, in XFCE or Gnome, the resolution is automatically fixed when the desktop environment starts.

This issue can be closed.

**Comment by andrewdavidwong** [comment]
> This issue can be closed.

Is there no code fix or documentation change needed? There mere existence of a workaround isn't always sufficient reason to close an issue.

**Comment by rapenne-s** [comment]
> > This issue can be closed.
> 
> Is there no code fix or documentation change needed? There mere existence of a workaround isn't always sufficient reason to close an issue.

Indeed. I'll make an OpenBSD bug report, we added support for virtio-gpu recently, not sure it's being used here, but at least we have active people knowing the graphical stack, so it might be fixable :)

**Comment by rapenne-s** [comment]
Well, OpenBSD uses by default the high resolution offered, which is 1280x768, but the QubeOS display window in practice has a size of 1280x720, so panning happens from OpenBSD PoV, which gives a vertical cursor mismatch between dom0 and OpenBSD because they expect different height.

Could it be possible to remove the 1280x768 resolution from QubesOS? This doesn't seem an OpenBSD bug here actually, the driver is just choosing a resolution that doesn't match the window, but it can't know about that.

**Comment by DemiMarie** [comment]
> Also, in XFCE or Gnome, the resolution is automatically fixed when the desktop environment starts.

Why is this?

> Could it be possible to remove the 1280x768 resolution from QubesOS? This doesn't seem an OpenBSD bug here actually, the driver is just choosing a resolution that doesn't match the window, but it can't know about that.

That is one possibility.  Making the window be 1280x720 is another.  That said, would it be possible to actually implement Qubes GUI support in OpenBSD?  I imagine it would not be _that_ hard to port once the necessary Xen support was added.  Alternatively, what about adding virtio-GPU support to the stubdomain?

**Comment by DemiMarie** [comment]
Still an issue.

**Comment by DemiMarie** [comment]
I have no clue if this is still an issue, and I don’t use OpenBSD anymore.
